_posts/2018-02-02-45-minutes-into-powerapps.md

```diff
@@ -0,0 +1,43 @@
+
+![](/img/powerapps/contacts-app-editor.png)
+
+Fiddling around with styles and nuances of UI is quite similar experience with all the other IDE’s we are accustomed to. Changing colors, fonts, borders, the usual stuff is straightforward, adding images and media needs a bit more imagination. Hit File -> Media -> Images. Then Browse your media files to upload them to service. After upload, you’ll see your files on the list and can give them name, or label if you will. From thereof you can use them as image elements' source by that label. Same goes for video and audio content as well.
+
+## Techinal things - The Klingon language part
+
+Personally I prefer Quenya and Telerin by the way but let's not get all excited. In Apple ecosystem's context PowerApps is an odd bird to say the least. Apple is notoriously jealous about what they allow to happen under their roof. Given the not-so-distant history of Microsoft's mobile endeavors it's not surprising to see them go to full lengths at utilising the potential of more successful platforms, namely Android and iOS. And who can blame them? It's exactly the right thing to do from business perspective. The only problem here is that they're not the ones holding all the cards in these turfs, so they need to adjust to the rules of others. Android ecosystem is pretty slack, though Google has started to tighten the reins with Oreo too. Apple on the other hand may pull the plug from this type of approach on a whim. Microsoft is not too keen on disclosing the details of how PowerApps are run on a device, but looking at the UX it's obvious that we're being served web-pages in a shell of a native app. Native as in ReactNative, beacuse even though they don't say it out loud, PowerApps Dashboard's Analytics tab gives unmistakeable clue on what is going on under the hood.
+
+![](/img/powerapps/powerapps-dashboard.png)
+
+This is not to say RN is inherently bad, or even that MS's apparoach is wrong. It's just something to keep in mind if you're planning on making PowerApps an integral part of your business processes.
+
+## Going public - In organisational context
+
+After tweaks and rebranding efforts it’s time to publish. Hit File -> Save and you’ll be one click away from gentle scrutiny of your first app by your colleagues.  First time around it’ll ask for who can do what with your app, options are Edit or View. Going back to start page of PowerApps and hitting Apps will give you list of the Apps available for you. At this point the App has magically appeared into your mobile devices PowerApps app too. from list selecting “…” -> Settings presents you with App’s global settings. From where you can edit various things about the app, but interesting bitsare the Share and Versions tabs. From Share you’ll be able to control who can do what with your app. From versions you control, which version of the app is currently published for the audience. 
```

Comment:
...with your app. Options are...
give you a list
has magically appeared into -> has magically transferred into
. from -> . From
presents you with your App's global
From where -> From there
bitsare -> bits are
control (pilkku pois) which version